aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -178,7 +180,7 @@ def _get_hashed_idempotency_key(self, data: Dict[str, Any]) -> str:
             warnings.warn(f"No value found for idempotency_key. jmespath: {self.event_key_jmespath}")
 
         generated_hash = self._generate_hash(data=data)
-        function_name = os.getenv(constants.LAMBDA_FUNCTION_NAME_ENV, "test-func")
+        function_name = os.getenv(constants.LAMBDA_FUNCTION_NAME_ENV, "test-func") + "." + self.function_name
```

Comment:
I was gonna call out that this will impact customers also using `@idempotency` instead of `@idempotency_function` only as reported... but then it ocurred to me customers using a single function with multiple adapters **and** idempotency will end up in the same bug.

suggestion to use f-string instead of multiple concatenation for performance and nitpick late on Monday evening

```suggestion
	lambda_fn_name = os.getenv(constants.LAMBDA_FUNCTION_NAME_ENV, "test-func")
	function_name = f"{lambda_fn_name}.{self.function_name}"
```

cc @cakepietoast for awareness